### ISSUE TITLE
Apache Pulsar Manager (APM) research

### ISSUE THREAD

**Issue by MPSebastian** [amosproj/amos2023ss01-apache-pulsar-ui#10]
## User story
1. As a PO
2. I want to know which of the requirements from our customer are already covered by [APM](https://github.com/apache/pulsar-manager)
3. So that I can decide on what our focus should be on.

## Acceptance criteria
* Use docker compose file to run a apache pulsar instance and the APM
* Experiment with the APM dashboard
* Create a document describing which requirements are already fulfilled by APM

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

**Comment by JulianTS** [comment]
Just a heads up for anyone trying it out. At least with the docker compose setup, I ran into the problem that i cannot get past the login page with the given _pulsar_ username and _pulsar_ password.

**Comment by iheziqi** [comment]
> Just a heads up for anyone trying it out. At least with the docker compose setup, I ran into the problem that i cannot get past the login page with the given _pulsar_ username and _pulsar_ password.

I also got stuck in this problem for an hour. Then I found out that we need to create a [super-user](https://github.com/apache/pulsar-manager#access-pulsar-manager) for APM version bigger than 0.1.0

**Comment by iheziqi** [comment]
CSRF_TOKEN=$(curl http://backend-service:7750/pulsar-manager/csrf-token)
curl \
    -H "X-XSRF-TOKEN: $CSRF_TOKEN" \
    -H "Cookie: XSRF-TOKEN=$CSRF_TOKEN;" \
    -H 'Content-Type: application/json' \
    -X PUT http://backend-service:7750/pulsar-manager/users/superuser \
    -d '{"name": "admin", "password": "apachepulsar", "description": "test", "email": "username@test.org"}'

**Comment by iheziqi** [comment]
For reference, I set up APM with docker-compose and couldn't create a new environment in APM using the service URL http://127.0.0.1:8080 suggested in the APM doc. This [issue](https://github.com/apache/pulsar-manager/issues/283#issuecomment-1033264016) helped me out. I managed to create a pulsar environment using http://pulsar:8080.
